packages/scene-system/sdk/Fetch.ts

```diff
@@ -33,16 +33,20 @@ export function createFetch({ canUseFetch, previewMode, log, originalFetch }: Fe
       return Promise.reject(new Error('This scene is not allowed to use fetch.'))
     }
 
-    async function fetchRequest() {
-      const abortController = new AbortController()
-      const timeout = setTimeout(() => {
-        abortController.abort()
-      }, init?.timeout || TIMEOUT_LIMIT)
-      const response = await originalFetch(resource, { signal: abortController.signal, ...init })
-      clearTimeout(timeout)
-      return response
-    }
-
-    return fifoFetch.add(fetchRequest)
+    return fifoFetch.add(() => fetchWithTimeout(originalFetch, resource, init))
   }
 }
+
+export async function fetchWithTimeout(
```

Comment:
can we move this function outside of scene-system? scene-system is another project and could be external repo and atomicsHelpers shouldn't depend on it

Reply:
@gonpombo8